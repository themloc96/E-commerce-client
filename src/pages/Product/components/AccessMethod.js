import React, { useEffect } from 'react';

function AccessMethod({ product }) {
  const {
    productDesigns,
    productAppServices,
    accessMethods,
    productPlusLinks,
  } = product;
  const defaultAccessMethodList = [
    { label: '스마트폰 App', value: '스마트폰 App' },
    { label: '지문인식', value: '지문인식' },
    { label: '스마트카드키', value: '스마트키카드' },
    { label: '스틱키', value: '스틱키' },
    { label: '비밀번호', value: '비밀번호' },
    { label: '기계식키', value: '기계식키' },
    { label: '얼굴인식', value: '얼굴인식' },
  ];
  const newAccessMethodData = [];
  defaultAccessMethodList.map(defaultMethod => {
    const accessMethodMatched = accessMethods.find(
      method => defaultMethod.label === method.value,
    );
    if (accessMethodMatched) {
      return newAccessMethodData.push(accessMethodMatched);
    }
    return newAccessMethodData;
  });

  return (
    <section className="access-method border-bottom">
      {accessMethods.length > 0 && (
        <div className="flex">
          <p
            className="f16Medium mr-[23px] lg:mr-[66px] lg:f20Medium access-method-title"
            style={{ lineHeight: '1' }}
          >
            출입방식
          </p>
          <ul className="access-method-items">
            {newAccessMethodData.map(item => {
              const { text, value, icon } = item;
              return (
                <li key={value}>
                  <div className="bg-bg2 rounded-[10px]">
                    <img src={icon} alt={text} />
                  </div>
                  <p className="mt-2 text-center f12Regular xl:f16Regular">
                    {text}
                  </p>
                </li>
              );
            })}
          </ul>
        </div>
      )}

      <div className="mt-[49px] mb-[18px] flex">
        <p className="font-medium txt-title-access-method mr-[23px] lg:mr-[66px]">디자인</p>
        <p className="txt-description-access-method">
          {productDesigns.toString().replace(/,/g, ', ')}
        </p>
      </div>
      <div className="mb-6 lg:mt-[30px] lg:mb-[33px] flex">
        <p className="font-medium txt-title-access-method mr-[23px] lg:mr-[66px]">App서비스</p>
        <p className="txt-description-access-method">
          {productAppServices.toString().replace(/,/g, ', ')}
        </p>
      </div>
      <div className="mb-6 lg:mt-[30px] lg:mb-[33px] flex">
        <p className="font-medium txt-title-access-method mr-[23px] lg:mr-[66px]">플러스 링크</p>
        <p className="txt-description-access-method">
          {productPlusLinks.toString().replace(/,/g, ', ')}
        </p>
      </div>
    </section>
  );
}

AccessMethod.propTypes = {};

export default AccessMethod;
