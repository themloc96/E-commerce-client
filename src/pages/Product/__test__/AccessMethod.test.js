import { createRenderer } from 'react-test-renderer/shallow';
import AccessMethod from '../components/AccessMethod';

const product = {
  productDesigns: ['text game'],
  productAppServices: ['text game'],
  accessMethods: [
    {
      text: '비밀번호',
      value: '비밀번호',
      icon: '/assets/icons/password-icon.png',
    },
  ],
};
const renderer = createRenderer();
describe('AccessMethod', () => {
  renderer.render(<AccessMethod product={product} />);
  const renderedOutput = renderer.getRenderOutput();
  it('Should render and match the snapshot', () => {
    expect(renderedOutput).toMatchSnapshot();
  });
});
