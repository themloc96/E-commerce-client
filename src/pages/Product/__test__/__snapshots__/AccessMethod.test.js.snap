// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessMethod Should render and match the snapshot 1`] = `
<section
  className="access-method border-bottom"
>
  <div
    className="flex"
  >
    <p
      className="f16Medium mr-[23px] lg:mr-[66px] lg:f20Medium"
      style={
        Object {
          "lineHeight": "1",
        }
      }
    >
      출입방식
    </p>
    <ul
      className="access-method-items"
    >
      <li>
        <div
          className="bg-bg2 rounded-[10px]"
        >
          <img
            alt="비밀번호"
            src="/assets/icons/password-icon.png"
          />
        </div>
        <p
          className="mt-2 text-center f12Regular xl:f16Regular"
        >
          비밀번호
        </p>
      </li>
    </ul>
  </div>
  <div
    className="mt-[49px] mb-[18px]"
  >
    <span
      className="f16Medium lg:f20Medium"
    >
      디자인
    </span>
    <span
      className="f16Regular lg:f20Medium ml-[60px] lg:ml-[85px]"
    >
      text game
    </span>
  </div>
  <div
    className="mb-6 lg:mt-[30px] lg:mb-[33px]"
  >
    <span
      className="f16Medium lg:f20Medium"
    >
      App서비스
    </span>
    <span
      className="f16Regular lg:f20Medium ml-[30px] lg:ml-[45px]"
    >
      text game
    </span>
  </div>
</section>
`;
